### ISSUE TITLE
Build failure for zombiezen.com/go/sqlite on windows-i386 and mips targets

### ISSUE THREAD

**Issue by tsynik** [anacrolix/torrent#711]
I have build issues of torrent with go 1.17.5 for windows-i386 and mips targets. Here is the log:

```
GOOS=windows GOARCH=386   go build -ldflags='-s -w' -o /Users/nikk/XCode/TorrServer/dist/TorrServer-windows-386.exe ./cmd
# zombiezen.com/go/sqlite
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:68:20: undefined: sqlite3.Xsqlite3session_create
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:90:2: undefined: sqlite3.Xsqlite3session_delete
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:725:37: undefined: sqlite3.SQLITE_CHANGESET_DATA
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:726:37: undefined: sqlite3.SQLITE_CHANGESET_NOTFOUND
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:727:37: undefined: sqlite3.SQLITE_CHANGESET_CONFLICT
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:728:37: undefined: sqlite3.SQLITE_CHANGESET_CONSTRAINT
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:729:37: undefined: sqlite3.SQLITE_CHANGESET_FOREIGN_KEY
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:761:33: undefined: sqlite3.SQLITE_CHANGESET_OMIT
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:765:34: undefined: sqlite3.SQLITE_CHANGESET_ABORT
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:779:36: undefined: sqlite3.SQLITE_CHANGESET_REPLACE
../../../go/pkg/mod/zombiezen.com/go/sqlite@v0.8.0/session.go:90:2: too many errors
GOOS=linux GOARCH=mips  GOMIPS=softfloat go build -ldflags='-s -w' -o /Users/nikk/XCode/TorrServer/dist/TorrServer-linux-mips ./cmd
package server/cmd
	imports server
	imports server/settings
	imports github.com/anacrolix/torrent
	imports github.com/anacrolix/torrent/storage
	imports zombiezen.com/go/sqlite
	imports modernc.org/libc
	imports modernc.org/libc/errno: build constraints exclude all Go files in /Users/nikk/go/pkg/mod/modernc.org/libc@v1.13.2/errno
package server/cmd
	imports server
	imports server/settings
	imports github.com/anacrolix/torrent
	imports github.com/anacrolix/torrent/storage
	imports zombiezen.com/go/sqlite
	imports modernc.org/libc
	imports modernc.org/libc/fcntl: build constraints exclude all Go files in /Users/nikk/go/pkg/mod/modernc.org/libc@v1.13.2/fcntl
...
```

**Comment by tsynik** [comment]
I can build those targets with -tags=nosqlite flag (as it doesn't needed for TorrServer) but maybe it's still the issue for others

**Comment by anacrolix** [comment]
Thanks for reporting this! Are you actively using those platforms or just building as part of some Matrix? Upstream might be interested to know that you're using sqlite with those platforms.

**Comment by tsynik** [comment]
> Thanks for reporting this! Are you actively using those platforms or just building as part of some Matrix? Upstream might be interested to know that you're using sqlite with those platforms.

Nope, I use arm and darwin, just building for others. There were users asked for i386 builds for old PC and users of Keenetic mips routers who wanted install TorrServer on it.

**Comment by anacrolix** [comment]
@tsynik see https://gitlab.com/cznic/sqlite/-/issues/86 and https://pkg.go.dev/modernc.org/sqlite?utm_source=godoc#hdr-Supported_platforms_and_architectures.

Perhaps adding the CGO implementation as a tag or default when CGO is enabled would work better. Unfortunately that probably requires writing some kind of shim between the crawshaw and zombiezen wrapper of modernc.org/sqlite that I use.

**Comment by tsynik** [comment]
@anacrolix TorrServer doesn't use torrent sqlite parts so we still can build all arch including mips and i386, so it's not an issue. For torrent, just need to mention that sqlite builds suppored only for x64 platforms excluding mips ;-)

**Comment by anacrolix** [comment]
@tsynik this should be fixed by 5a61d8f6ac937636f8253b254bf1be0c4cca0621.
